R/utils.R

```diff
@@ -45,3 +47,35 @@ prepareArgs <- function (args) {
         NULL
     }
 }
+
+switchFieldNamesWithTitles <- function (id, env, res) {
+    # Retrieve the list of ordered fields
+    ds_fields <- do.call(Dataset.fields, list(id, limit=1000, env=env))$data
+
+    # Create a dataframe that maps column names with titles
+    if ("_id" %in% names(res)) {
+        # Append column "_id" to the list of names and titles because it's always present in result query, but not in names and titles
+        col.name.title.map <- data.frame(
+            names = c(ds_fields$name, "_id"),
+            title = c(ds_fields$title, "_ID"),
+            stringsAsFactors = FALSE
+        )
+    }
+    else {
+        col.name.title.map <- data.frame(
+            names = ds_fields$name,
+            title = ds_fields$title,
+            stringsAsFactors = FALSE
+        )
+    }
+    # Remove all names and titles that are not in the results query
+    col.name.title.map <- col.name.title.map[col.name.title.map$names %in% colnames(res), ]
+
+    # Order columns in the dataframe based on list of dataset fields
+    res <- res[col.name.title.map$names]
```

Comment:
The fields should still be ordered even if we aren't mapping names to titles.

Reply:
Do you think it should be ordered by the list of fields specified by the user here: Dataset.query(ds$id, fields<- list(...)), or by the default order returned from the Dataset.fields() ? Current implementation is the second one.

Reply:
It should do the second one (use the ordering stored in the API), users can always reorder after querying.